packages/main/src/server/cache-manager.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CachedContent, Content, Part, RequestOptions, Tool, ToolConfig } from "../../types";
+import {
+  CachedContentUrl,
+  FilesRequestUrl,
+  getHeaders,
+  makeServerRequest,
+} from "./request";
+import {
+  CachedContentCreateParams,
+  CachedContentUpdateParams,
+  ListCacheResponse,
+  ListParams,
+} from "../../types/server";
+import { RpcTask } from "./constants";
+import { GoogleGenerativeAIError } from "../errors";
+
+/**
+ * Class for managing GoogleAI file uploads.
+ * @public
+ */
+export class GoogleAICacheManager {
+  model: string;
+  ttl?: string;
+  tools?: Tool[];
+  toolConfig?: ToolConfig;
+  systemInstruction?: string | Part | Content;
+
+  constructor(
+    public apiKey: string,
+    private _requestOptions: RequestOptions,
+  ) {}
+
+  /**
+   * Upload a new content cache
+   */
+  async create(
+    createOptions: CachedContentCreateParams,
+  ): Promise<CachedContent> {
+    const newCachedContent: CachedContent = { ...createOptions };
+    if (createOptions.ttlSeconds) {
+      newCachedContent.ttl = createOptions.ttlSeconds.toString() + "s";
+      delete (newCachedContent as CachedContentCreateParams).ttlSeconds;
+    }
+    if (!newCachedContent.model.includes("/")) {
```

Comment:
What if model is empty, shouldn't we report an error? Or do we leave that up to the service?

Reply:
Seems like a good check, added.